features/dd-sdk-android-trace/src/main/kotlin/com/datadog/android/trace/impl/internal/DatadogTraceIdAdapter.kt

```diff
@@ -8,7 +8,7 @@ package com.datadog.android.trace.impl.internal
 import com.datadog.android.trace.api.trace.DatadogTraceId
 import com.datadog.trace.api.DDTraceId
 
-internal class DatadogTraceIdAdapter(private val delegate: DDTraceId) : DatadogTraceId, DDTraceId() {
+internal data class DatadogTraceIdAdapter(private val delegate: DDTraceId) : DatadogTraceId, DDTraceId() {
```

Comment:
do we need it to be `data` class?

Reply:
`data` class force kotlin compiler generate `hashCode` and `equals` methods, that take into account every property declared in constructor, which required in tests, allowing to do such comparison: 

```kotlin 
assertThat(testedAgentSpanContextAdapter.traceId).isEqualTo(DatadogTraceIdAdapter(ddTraceId))
```